flask_restful/__init__.py

```diff
@@ -272,15 +272,21 @@ def handle_error(self, e):
         """
         got_request_exception.send(current_app._get_current_object(), exception=e)
 
-        if not hasattr(e, 'code') and current_app.propagate_exceptions:
+        if not isinstance(e, HTTPException) and current_app.propagate_exceptions:
             exc_type, exc_value, tb = sys.exc_info()
             if exc_value is e:
                 raise
             else:
                 raise e
 
-        code = getattr(e, 'code', 500)
-        data = getattr(e, 'data', error_data(code))
+        if isinstance(e, HTTPException):
+            code = e.code
+            data = getattr(e, 'data', error_data(code))
```

Comment:
Changing this causes 4 failures in the test suite. Data is directly assigned to the Mock object: https://github.com/flask-restful/flask-restful/blob/master/tests/test_api.py#L445

I am fine with removing this as it seems to be a "feature" added only by the tests, but want to make sure before I do.